Library/Homebrew/utils.rb

```diff
@@ -88,8 +88,10 @@ def ohai(title, *sput)
   puts sput
 end
 
-def oh1(title)
-  title = Tty.truncate(title) if $stdout.tty? && !ARGV.verbose?
+def oh1(title, options = {})
+  if $stdout.tty? && !ARGV.verbose? && options.fetch(:truncate, true)
+    title = Tty.truncate(title)
+  end
```

Comment:
One reason why I suggested to name the option `:allow_truncation` (though that's a bit unwieldy) is that it's less confusing than a plain `:truncate` with a Boolean value. Example:

```ruby
oh1 "This should always be truncated", :truncate => true # default
oh1 "This should never be truncated", :truncate => false
```

I would expect the first line to be always truncated, but instead it means “may truncate”, depending on environmental factors like output to a TTY or the presence of the `--verbose` flag. One idea I have to resolve this is to do the following (though I'm happy to hear about other/better ideas):

```ruby
oh1 "This will sometimes be truncated", :truncate => :auto # default
oh1 "This should never be truncated", :truncate => :never
```